packages/ptokens-assets-evm/src/ptokens-evm-asset-builder.ts

```diff
@@ -30,25 +30,21 @@ export class pTokensEvmAssetBuilder extends pTokensAssetBuilder {
       method: 'hub',
       abi: factoryAbi,
     })
-
-    if (this._assetInfo.assetTokenAddress !== this._assetInfo.underlyingAssetTokenAddress) {
-      const pTokenAddress = await this._provider.makeContractCall<string, [string, string, number, string, string]>(
-        {
-          contractAddress: this.factoryAddress,
-          method: 'getPTokenAddress',
-          abi: factoryAbi,
-        },
-        [
-          this.assetInfo.underlyingAssetName,
-          this.assetInfo.underlyingAssetSymbol,
-          this.assetInfo.underlyingAssetDecimals,
-          this.assetInfo.underlyingAssetTokenAddress,
-          this.assetInfo.underlyingAssetNetworkId,
-        ],
-      )
-      if (!this.assetInfo.assetTokenAddress) this.assetInfo.assetTokenAddress = pTokenAddress
-      else if (pTokenAddress !== this.assetInfo.assetTokenAddress) throw new Error('Invalid pToken address')
-    }
+    const pTokenAddress = await this._provider.makeContractCall<string, [string, string, number, string, string]>(
```

Comment:
factorize this into a function, easier to test

Reply:
not implemented yet

Reply:
done